sky/resources.py

```diff
@@ -41,17 +41,15 @@ class Resources:
         # TODO:
         sky.Resources(requests={'mem': '16g', 'cpu': 8})
     """
-    # If any fields changed:
-    # 1. Increment the version. For backward compatibility.
-    # 2. Change the __setstate__ method to handle the new fields.
-    # 3. Modify the to_config method to handle the new fields.
+    # If any fields changed, increment the version. For backward compatibility,
+    # modify the __setstate__ method to handle the old version.
```

Comment:
We do need the step of changing `from_yaml_config()` too?

Reply:
I don't think it's particularly relevant to backward compatibility, do you?

Reply:
Makes sense. Was thinking this comment is a guide for "all the places we need to change when we add a new field". For just back compat, it seems `__setstate__` is the only required change. cc @Michaelvll to double check

Reply:
I think it's hard to provide such a comprehensive guideline, because in the worst case almost all the methods would have to be fixed. It's her responsibility to add necessary changes when a new field is added.